_drafts/2023-05-11-ota-for-embedded-linux-devices.md

```diff
@@ -0,0 +1,446 @@
+---
+title: "OTA for Embedded Linux Devices: A practical introduction"
+author: tsarlandie
+tags: [linux, ota, swupdate]
+---
+
+<!-- excerpt start -->
+
+A core belief of Memfault is that we can ship faster when we have efficient
+tools to detect, debug and fix issues in the field. Fixing issues on your
+computer is not enough: the fix must reach the end users!
+
+<!-- excerpt end -->
+
+A good OTA setup should allow you to quickly prepare updates and ship them with
+confidence. In particular:
+
+- It should be predictable: systems should be identical (ideally, exact
+  byte-level copies) regardless of whether they are a fresh install or have been
+  updated 10 times in their lifespan;
+- It should be reliable: interruptions of network or power should not risk
+  damaging the device,
+- It should be resilient so it can still update devices even when they are in a
+  non-nominal state
+
+As you will see, many components collaborate to update your embedded Linux
+systems. Our goal in this blog post is to walk you through a complete update
+cycle, introducing the different components and showing how they interact with
+each other.
+
+For today’s discussion, we will focus on a system built with Yocto, booting on
+an A/B partition scheme, with [U-Boot](https://u-boot.readthedocs.io/en/latest/)
+as the bootloader and [SWUpdate](https://sbabic.github.io/swupdate/index.html)
+as the updater. This is an extremely common setup, completely open-source (free
+as beer and speech), and the one we use in Memfault Linux SDK examples. But the
+concepts shown here will apply to a lot of other OTA strategies as well and it’s
+important to point out that Memfault for Linux is not limited to this setup.
+
+> **Understanding Yocto and the A/B partitioning scheme**
```

Comment:
I would probably just move this to it's own section in the beginning of the article.

I would also add a Table of Contents (look in a few of the other posts to figure out how)

Reply:
This is information that is unrelated to the core points of the articles. I made it an aside so that it would be clear that this is not what the article is about. I am afraid we would lose readers otherwise. 

Added TOC. 🙏